### PR TITLE
docs(flutter): Use sentry flutter init in samples

### PR DESCRIPTION
**Pull request by denrase**
## DESCRIBE YOUR PR
*Tell us what you're changing and why. If your PR **resolves an issue**, please link it so it closes automatically.*

Add samples that use `SentryFlutter.init` instead of `Sentry.init`.

Relates to https://github.com/getsentry/sentry-dart/issues/2383

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [x] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## EXTRA RESOURCES

- [Issue](https://github.com/getsentry/sentry-dart/issues/2383)

### CODE REVIEW

File: platform-includes/configuration/allow-urls/flutter.mdx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+```dart
+import 'package:flutter/widgets.dart';
+import 'package:sentry_flutter/sentry_flutter.dart';
+
+Future<void> main() async {
```
Comment (buenaflor) [inline platform-includes/configuration/allow-urls/flutter.mdx:5]:
```suggestion
```

imo we can omit the main() function and the import for all future code snippets (except for snippets in the getting started page), they are just a lot of noise

File: platform-includes/configuration/allow-urls/flutter.mdx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+```dart
```
Comment (buenaflor) [inline platform-includes/configuration/allow-urls/flutter.mdx:1]:
```suggestion
```dart {3}
```

let's add code highlighting for the affected lines, line 3 is probably not correct just an example

File: platform-includes/configuration/allow-urls/flutter.mdx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import 'package:flutter/widgets.dart';
+import 'package:sentry_flutter/sentry_flutter.dart';
+
+Future<void> main() async {
+  await SentryFlutter.init(
+    (options) {
+      options.dsn = '___PUBLIC_DSN___';
```
Comment (buenaflor) [inline platform-includes/configuration/allow-urls/flutter.mdx:8]:
```suggestion
```

imo we can leave dsn out (and in the other inits as well)

**Review by buenaflor — CHANGES_REQUESTED** [review]
left some comments, I think we can improve the code snippets in general to reduce the size of the snippets

**Comment by denrase** [comment]
@buenaflor I noticed some formatting issues and one codeblock is still using Sentry.init. Will udate and notify you.

File: platform-includes/configuration/before-send-fingerprint/flutter.mdx [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  }
+  return event;
+}
+
+await Sentry.init(
+  (options) {
+    options.beforeSend = beforeSend;
+  },
+);
+```
```
Comment (buenaflor) [inline platform-includes/configuration/before-send-fingerprint/flutter.mdx:14]: let's keep it concise, don't create another function but do it inline e.g

```dart
options.beforeSend = (event, hint) {
	// etc..
	return event;
}
```

File: platform-includes/configuration/before-send-hint/flutter.mdx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+```dart {7}
+FutureOr<SentryEvent?> beforeSend(SentryEvent event, Hint hint) async {
+  return hint is MyHint ? null : event;
+}
+
+await SentryFlutter.init(
+  (options) {
+    options.beforeSend = beforeSend;
+  },
+);
```
Comment (buenaflor) [inline platform-includes/configuration/before-send-hint/flutter.mdx:10]: same as the other about keeping concise

File: platform-includes/configuration/before-send/flutter.mdx [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  event = event.copyWith(serverName: null); // Don't send server names.
+  return event;
+}
+
+await SentryFlutter.init(
+  (options) {
+    options.beforeSend = beforeSend;
+  },
+);
+```
```
Comment (buenaflor) [inline platform-includes/configuration/before-send/flutter.mdx:13]: same as the other about keeping concise

File: platform-includes/configuration/sample-rate/flutter.mdx [hunk 7]
```diff
@@ -0,0 +1,9 @@
+```dart
+import 'package:flutter/widgets.dart';
+import 'package:sentry_flutter/sentry_flutter.dart';
+
+Future<void> main() async {
+  // Capture only 25% of events
+  await SentryFlutter.init((options) => options.sampleRate = 0.25);
+}
+```
```
Comment (buenaflor) [inline platform-includes/configuration/sample-rate/flutter.mdx:9]: do we need the import and add void main? at this point the user should've gone through the intro anyway, imo I'd prefer this and highlight the sampleRate line

```dart
await SentryFlutter.init(
  (options) {
		// Capture only 25% of events
    options.sampleRate = 0.25;
  },
);
```

wdyt

File: platform-includes/enriching-events/attach-viewhierarchy/flutter.mdx [hunk 8]
```diff
@@ -1,14 +1,8 @@
...
-    },
-    appRunner: () => runApp(MyApp()),
-  );
-}
+```dart {3}
+await SentryFlutter.init(
+  (options) {
+    options.attachViewHierarchy = true;
+  },
+  appRunner: () => runApp(MyApp()),
```
Comment (buenaflor) [inline platform-includes/enriching-events/attach-viewhierarchy/flutter.mdx:6]:
```suggestion
```

don't think we need it necessarily

**Review by buenaflor — CHANGES_REQUESTED** [review]
I didn't add comment everywhere but if possible inline the callbacks to make the code snippets a bit more digestible

File: platform-includes/user-feedback/sdk-api-example/flutter.mdx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+final userFeedback = SentryUserFeedback(
+    eventId: sentryId,
+    comments: 'Hello World!',
+    email: 'foo@bar.org',
+    name: 'John Doe',
+);
+
+Sentry.captureUserFeedback(userFeedback);
+```
```
Comment (buenaflor) [inline platform-includes/user-feedback/sdk-api-example/flutter.mdx:26]: will we need this? I think we are going to update the feedback docs anyway: https://github.com/getsentry/sentry-docs/issues/11756

Comment (denrase) [reply platform-includes/user-feedback/sdk-api-example/flutter.mdx:26]: just added it to be in line with the others. we can remove/update then once the other docs land

File: platform-includes/configuration/sample-rate/dart.mdx [hunk 10]
```diff
@@ -1,8 +1,4 @@
 ```dart
-import 'package:sentry/sentry.dart';
-
-Future<void> main() async {
-  // Capture only 25% of events
-  await Sentry.init((options) => options.sampleRate = 0.25);
-}
+// Capture only 25% of events
+await Sentry.init((options) => options.sampleRate = 0.25);
```
Comment (buenaflor) [inline platform-includes/configuration/sample-rate/dart.mdx:3]:
```suggestion
await Sentry.init((options) {
  // Capture only 25% of events
  options.sampleRate = 0.25;
});
```

and highlight pls

File: platform-includes/performance/traces-sample-rate/dart.mdx [hunk 11]
```diff
@@ -1,7 +1,7 @@
...
-import 'package:sentry/sentry.dart';
-
-Sentry.init((options) => {
-      options.tracesSampleRate = 0.2,
-    });
+Sentry.init(
+  (options) {
+    options.tracesSampleRate = 0.2;
+  },
+);
```
Comment (buenaflor) [inline platform-includes/performance/traces-sample-rate/dart.mdx:6]:
```suggestion
await Sentry.init((options) {
  options.tracesSampleRate = 0.2;
});
```

File: platform-includes/set-fingerprint/basic/dart.mdx [hunk 12]
```diff
@@ -1,14 +1,12 @@
...
+await SentryFlutter.init(
+  (options) {
+    options.beforeSend = (event, hint) {
+      if (event.throwable is DatabaseException) {
+        event = event.copyWith(fingerprint: ['database-connection-error']);
+      }
+      return event;
+    };
+  },
+);
```
Comment (buenaflor) [inline platform-includes/set-fingerprint/basic/dart.mdx:11]:
```suggestion
```dart {2-6}
await SentryFlutter.init((options) {
  options.beforeSend = (event, hint) {
    if (event.throwable is DatabaseException) {
      event = event.copyWith(fingerprint: ['database-connection-error']);
    }
    return event;
  };
});
```

File: platform-includes/enriching-events/attach-viewhierarchy/flutter.mdx [hunk 13]
```diff
@@ -1,14 +1,7 @@
...
-    },
-    appRunner: () => runApp(MyApp()),
-  );
-}
+```dart {3}
+await SentryFlutter.init(
+  (options) {
+    options.attachViewHierarchy = true;
+  },
+);
```
Comment (buenaflor) [inline platform-includes/enriching-events/attach-viewhierarchy/flutter.mdx:6]:
```suggestion
```dart {2}
await SentryFlutter.init((options) {
	options.attachViewHierarchy = true;
});
```

**Review by buenaflor — CHANGES_REQUESTED** [review]
one last change, let's make the code a bit more readable but reducing the newlines a bit, pls do it everywhere else as well 🙏

**Comment by denrase** [comment]
@buenaflor Now we should have the same rules applied to most of the dart/flutter samples. We should klick all of them through  when vercel is deployed to see whether everything works as expected.

File: platform-includes/user-feedback/sdk-api-example/flutter.mdx (hunk 9 above)
Comment (buenaflor) [reply platform-includes/user-feedback/sdk-api-example/flutter.mdx]: okay then let's change this when we're doing the new user feedback

**Review by buenaflor — APPROVED** [review]

**Comment by buenaflor** [comment]
@denrase preview is deployed

**Pull request merged**